File: sem3/dsa/sorting/selection-sort.py
def exchange(array, i, j):
    temp = array[i]
    array[i] = array[j]
    array[j] = temp

def selection_sort(array):
    for i in range(len(array)):
        min_item = i
        for j in range(i+1, len(array)):
            if array[j] < array[min_item]:
                min_item = j
        exchange(array, i, min_item)
    return array

if __name__ == "__main__":
    a1, a2 = [4,2,2,4,0,1], [0, 0]
    selection_sort(a1), selection_sort(a2)
    assert a1 == [0,1,2,2,4,4]
    assert a2 == [0, 0]

File: sem3/dsa/sorting/selection_sort.c
#include <stdio.h>

void exchange(int i, int j, int array[]);

void selection_sort(int array[], int N){
  for (int i=0; i<N; i++){
    int a_min = i;
    for(int j=i+1; j<N; j++)
      if (array[j] < array[a_min])
        a_min = j;
    exchange(i, a_min, array);
  }
}

void exchange(int i, int j, int array[]){
  int temp = array[i];
  array[i] = array[j];
  array[j] = temp;
}

int main(){
  int array[] = {5,3,0,4,2,5};
  int N = sizeof(array)/sizeof(int);
  selection_sort(array, N);
  for (int i=0; i<N; i++){
    printf("%d ", array[i]);
  }
}
